.github/compat/long_compat.csv

```diff
@@ -1,4 +1,5 @@
 runner,compiler,version,support
+macos-13,intel-classic,2021.5,&check;runner,compiler,version,support
```

Comment:
Only the first line of this file should have headers.

In general when we add support for new toolchains/versions we need to

* update `matrix.yml` with new version combinations we expect to support
* manually trigger the CI in reporting mode (on your PR branch on your fork)
* get the diffs for `long_compat.csv`, `compat.csv`, and `README.md`, and splice them into the PR
* make sure CI passes again in test mode before the PR merges

Maybe we should add a developer doc with info on this workflow, I can do that.

Reply:
Documentation on this would be great! How should mkl be included, in the same table or in a separate one? Could be a bit too much information at once if it's in the same table.